Upgrade PSI to 1.2-RC1 conclave release

snedamle opened the pull request.

snedamle committed:
Upgrade PSI to 1.2-RC1 conclave release

shamsasari commented on psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java:
```diff
@@ -2,130 +2,148 @@
+                do
+                {
+                    //poll for reply to enclave
+                    responseMail = enclaveClient.pollMail();
```
You need to add a call to `Thread.sleep`, otherwise the client will be spamming the web server. 5-10 sec delay should be fine.

snedamle replied:
done

shamsasari replied:
Minor point, but it will be better to have the sleep before the `pollMail`. Because if `pollMail` did actually return a Mail object then you still have to wait 10s to get it!

snedamle replied:
done

shamsasari commented on psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java:
```diff
@@ -2,130 +2,148 @@
-        output.close();
-        return output;
+    public static void main(String... args) {
+        int exitCode = new CommandLine(new Client()).execute(args);
```
You should call `System.exit(exitCode);`.

snedamle replied:
done

shamsasari commented on psi-sample/host/build.gradle:
```diff
@@ -1,18 +1,26 @@
 
     // We need the enclave at runtime but not when compiling the host, because they don't share types.
     runtimeOnly project(path: ":enclave", configuration: mode)
     runtimeOnly "org.slf4j:slf4j-simple:1.7.30"
```
This isn't needed anymore as `conclave-web-host` already adds it

snedamle replied:
you mean runtimeOnly "org.slf4j:slf4j-simple:1.7.30" right?

shamsasari commented on psi-sample/gradle.properties:
```diff
@@ -1,4 +1,4 @@
 #
 #Tue Feb 09 22:52:29 GMT 2021
```
Probably remove this

snedamle replied:
done

shamsasari commented on psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java:
```diff
@@ -2,130 +2,148 @@
+                converter = EnclaveConstraintConverter.class)
+        private EnclaveConstraint constraint;
+
+        @CommandLine.Option(names = {"-f", "--file-state"},
+                required = true,
+                description = "File to store the state of the client. If the file doesn't exist a new one will be created.")
+        private Path file;
```
As far as I can see, your sample doesn't need a client state file. We can discuss offline.

snedamle replied:
this sample does not require state save. hence removing the file option from command line and the file check in client code.

snedamle committed:
code review updates.
this sample does not require you to reload an old state. removing the file argument from command line and the check in client code.
updating readme to reflect these changes.

shamsasari commented on psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java:
```diff
@@ -2,130 +2,135 @@
+
+            //responseMail is null till enclave doesn't reply back to the client
+            if(responseMail == null) {
+                do
+                {
```
Minor: Move the `{` up to the `do` line, to be consistent with the rest of the code.

snedamle replied:
done

shamsasari commented on psi-sample/host/build.gradle:
```diff
@@ -1,18 +1,25 @@
+def mode = findProperty("enclaveMode")?.toString()?.toLowerCase() ?: "mock"
 
 dependencies {
     implementation "com.r3.conclave:conclave-host:$conclaveVersion"
```
You don't need this either.

snedamle replied:
done

shamsasari commented on psi-sample/README.md:
```diff
@@ -51,72 +51,98 @@ Merchant - Merchant passes credit card numbers of all users who have made a purc
 
-On your terminal, start Service Provider and pass in the credit card numbers of users who have clicked the ad
+We will package the host jar into a big fat jar containing all dependencies using the shadow jar plugin. Add this to 
+hosts build.gradle. Note that 
```
"Note that"?

snedamle replied:
done

shamsasari commented on psi-sample/README.md:
```diff
@@ -51,72 +51,98 @@ Merchant - Merchant passes credit card numbers of all users who have made a purc
 
-    ./gradlew client:run --args="SERVICE-PROVIDER <CONSTRAINT> 88988898 77879 00989"
+    plugins {
+    id 'com.github.johnrengelman.shadow' version '6.1.0'
```
indent

snedamle replied:
done

shamsasari commented on psi-sample/README.md:
````diff
@@ -51,72 +51,98 @@ Merchant - Merchant passes credit card numbers of all users who have made a purc
-Enclave code hash:   26037FC0370589FEA489110ED8124223650F5620B0732F94CDDEDDF207F07457
-Enclave code signer: 4502FEF2B5973A9DCF2F5C85358ED9F099C7738300364A7D7451371E43694A85
-```
+Add conclave-client dependency to the client build.gradle. This consists of the EnclaveClient class which will be used 
````
conclave-web-client dependency

snedamle replied:
done

shamsasari commented on psi-sample/README.md:
```diff
@@ -51,72 +51,98 @@ Merchant - Merchant passes credit card numbers of all users who have made a purc
+
+#### A note about enclave constraint
+In this sample the `code signer` is used as enclave constraint, but you can also use the `code hash`. 
+If you want to use it, remember to change the code of the client to:
+
+`EnclaveConstraint.parse("C:"+ constraint +" SEC:INSECURE" ).check(attestation);`
```
You don't change the code, but rather simply pass in a different arg to the `--constraint` flag.

snedamle replied:
done

shamsasari commented on psi-sample/README.md:
```diff
@@ -51,72 +51,98 @@ Merchant - Merchant passes credit card numbers of all users who have made a purc
+
+`EnclaveConstraint.parse("C:"+ constraint +" SEC:INSECURE" ).check(attestation);`
+
+Read more in the [documentation](https://docs.conclave.net/enclave-configuration.html#productid).
```
Why is this referencing the product ID docs in this section? Doesn't seem to belong here.

snedamle replied:
done changed to the correct docs

shamsasari commented on psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java:
```diff
@@ -2,130 +2,135 @@
+        System.out.println("string.length" + creditCardNumbers.size());
+        System.out.println("string.length" + creditCardNumbers.get(0));
+
+        if (MERCHANT.equals(creditCardNumbers.get(0))) {
```
It will be better and clearer to have a separate flag called `--role` (or whatever term is best) for the merchant/service provider difference. You can define an enum for it an picocli will automatically deal with it.
```
enum class Role {
  MERCHANT,
  SERVICE_PROVIDER
}
```

Having it in the credit card numbers list will be confusing to the developer

snedamle replied:
done

snedamle committed:
code review updates.
passing merchant and service-provider as enum values
updated readme

shamsasari commented on psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java:
```diff
@@ -2,130 +2,140 @@
+    @CommandLine.Option(names = {"-u", "--url"},
+                required = true,
+                description = "URL of the web host running the enclave.")
+        private String url;
```
You can use Intellij to auto indent and format your code.

snedamle replied:
done

shamsasari commented on psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java:
```diff
@@ -2,130 +2,140 @@
+
+    @Override
+    public Void call() throws Exception {
+        //a new private key is generated. Enclave Client is created using this private key and constraint.
```
Comments should always begin with a capital letter.

snedamle replied:
done

shamsasari commented on psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java:
```diff
@@ -2,130 +2,140 @@
-            userDetailsList = new ArrayList<>(args.length);
-            for (int i =2; i< args.length; i++) {
-                UserDetails userDetails = new UserDetails(args[i]);
+        if (Role.MERCHANT.name().equals(role.name())) {
```
`role == Role.MERCHANT` 

This is how enums are used. Likewise for below.

snedamle replied:
done

shamsasari commented on psi-sample/enclave/src/main/java/com/r3/conclave/sample/enclave/PSIEnclave.java:
```diff
@@ -24,7 +24,7 @@
 
     private static final String MERCHANT = "MERCHANT";
-    private static final String SERVICE_PROVIDER = "SERVICE-PROVIDER";
+    private static final String SERVICE_PROVIDER = "SERVICE_PROVIDER";
```
To showcase good Conclave app design, you should move the `Role` enum to the `common` module and then use it in the Enclave as well, rather than having these constants.

snedamle replied:
done

snedamle committed:
moving Role enum to common

snedamle committed:
1. removed kryo
2. added Role enum for client type
3. added serialization-config.json
4. applied some sonar fixes

shamsasari commented on psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java:
```diff
@@ -1,131 +1,132 @@
+
+	@CommandLine.Option(names = {"-r", "--role"}
+			, description = "Role Options: ${MERCHANT/SERVICE_PROVIDER}")
+	private static Role role = null;
```
Remove the `static`

snedamle replied:
done

snedamle committed:
improved comments, removed Kryo dependency, removed tab indentation

snedamle committed:
removing static

shamsasari reviewed (approved):
👍

snedamle committed:
updating version to 1.2. tested with conclave-sdk-1.2
